packages/core/src/route/types.ts

```diff
@@ -84,12 +90,19 @@ export interface IPropDecoratorOptions {
 	typeFactory?: TypeValueFactory;
 	required?: boolean;
 }
-
-export type IPropDecoratorOptionsFactory = () => Maybe<IPropDecoratorOptions>;
-
-export interface IPropDecoratorMetadata {
-	key: string;
-	optsFactory?: IPropDecoratorOptionsFactory;
+export type IPropDecoratorOptionsFactory = IDecoratorOptionsFactory<IPropDecoratorOptions>;
+export type IPropDecoratorMetadata = IDecoratorMetadata<IPropDecoratorOptions>;
+
+export type IPropDependsDecoratorOptions = string[];
```

Comment:
I don't think that a list of strings is always the case here
<img width="788" alt="Screenshot 2020-12-02 at 15 44 54" src="https://user-images.githubusercontent.com/2676481/100895353-5bdf0700-34b5-11eb-83fe-cabe9e847e3e.png">